test/unit/StakingPool/depositTo.js

```diff
@@ -0,0 +1,774 @@
+const { ethers } = require('hardhat');
+const { expect } = require('chai');
+const { getTranches, getNewRewardShares, estimateStakeShares, POOL_FEE_DENOMINATOR } = require('./helpers');
+const { setEtherBalance, increaseTime } = require('../../utils/evm');
+const { BigNumber } = require('ethers');
+
+const { AddressZero } = ethers.constants;
+const { parseEther } = ethers.utils;
+
+describe('depositTo', function () {
+  const depositToFixture = {
+    poolId: 0,
+    initialPoolFee: 5, // 5%
+    maxPoolFee: 5, // 5%
+    productInitializationParams: [
+      {
+        productId: 0,
+        weight: 100,
+        initialPrice: 500,
+        targetPrice: 500,
+      },
+    ],
+    amount: parseEther('100'),
+    trancheId: 0,
+    tokenId: 0,
+    destination: AddressZero,
+    depositNftId: 1,
+  };
+
+  before(async function () {
+    const { stakingPool, cover } = this;
+    const { defaultSender: manager } = this.accounts;
+
+    const { poolId, initialPoolFee, maxPoolFee, productInitializationParams } = depositToFixture;
+
+    const coverSigner = await ethers.getImpersonatedSigner(cover.address);
+    await setEtherBalance(coverSigner.address, ethers.utils.parseEther('1'));
+    this.coverSigner = coverSigner;
+
+    await stakingPool
+      .connect(coverSigner)
+      .initialize(manager.address, false, initialPoolFee, maxPoolFee, productInitializationParams, poolId);
+  });
+
+  it('reverts if caller is not cover contract or manager when pool is private', async function () {
+    const { stakingPool, cover } = this;
+    const {
+      members: [user],
+      defaultSender: manager,
+    } = this.accounts;
+
+    const { amount, trancheId, tokenId, destination } = depositToFixture;
+
+    await stakingPool.connect(manager).setPoolPrivacy(true);
+
+    await expect(
+      stakingPool.connect(user).depositTo([
+        {
+          amount,
+          trancheId,
+          tokenId,
+          destination,
+        },
+      ]),
+    ).to.be.revertedWith('StakingPool: The pool is private');
+
+    const coverSigner = await ethers.getImpersonatedSigner(cover.address);
+
+    await expect(
+      stakingPool.connect(coverSigner).depositTo([
+        {
+          amount,
+          trancheId,
+          tokenId,
+          destination,
+        },
+      ]),
+    ).to.not.be.revertedWith('StakingPool: The pool is private');
+
+    await expect(
+      stakingPool.connect(manager).depositTo([
+        {
+          amount,
+          trancheId,
+          tokenId,
+          destination,
+        },
+      ]),
+    ).to.not.be.revertedWith('StakingPool: The pool is private');
```

Comment:
i would just make this to.not.be.reverted

let's make a successful call

Reply:
Updated the test here [Improve private pool unit test](https://github.com/NexusMutual/smart-contracts/pull/495/commits/6b6b6e6997dbac0f76f47f53392357be9409c785)